src/main/java/de/tum/cit/aet/artemis/buildagent/service/BuildJobExecutionService.java

```diff
@@ -79,6 +87,38 @@ public BuildJobExecutionService(BuildJobContainerService buildJobContainerServic
         this.buildLogsMap = buildLogsMap;
     }
 
+    /**
+     * This method is responsible for cleaning up temporary directories that were used for checking out repositories.
+     * It is triggered when the application is ready and runs asynchronously.
+     */
+    @EventListener(ApplicationReadyEvent.class)
+    @Async
+    public void initAsync() {
+        final ZonedDateTime currentTime = ZonedDateTime.now();
+        cleanUpTempDirectoriesAsync(currentTime);
+    }
+
+    private void cleanUpTempDirectoriesAsync(ZonedDateTime currentTime) {
+        log.info("Cleaning up temporary directories in {}", CHECKED_OUT_REPOS_TEMP_DIR);
+        try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(Path.of(CHECKED_OUT_REPOS_TEMP_DIR))) {
+            for (Path path : directoryStream) {
+                try {
+                    ZonedDateTime lastModifiedTime = ZonedDateTime.ofInstant(Files.getLastModifiedTime(path).toInstant(), currentTime.getZone());
+                    if (Files.isDirectory(path) && lastModifiedTime.isBefore(currentTime.minus(TEMP_DIR_RETENTION_PERIOD))) {
```

Comment:
Even though the system is supposed to only store directories in the `CHECKED_OUT_REPOS_TEMP_DIR`, we could make the system even more robust by also deleting old files, right? Do you see a problem with that?

Reply:
As CodeRabbit would say, this is "Nitpick" ^^

Reply:
Discussed: Let's keep files for debug reasons

Reply:
There should only be directories there. So if there was a file I would rather keep for debugging purposes